website/photos/services.py

```diff
@@ -1,8 +1,22 @@
+import hashlib
+import logging
+import os
+import tarfile
+from zipfile import ZipInfo, is_zipfile, ZipFile
+
+from django.conf import settings
+from django.contrib import messages
+from django.core.files import File
 from django.db.models import When, Value, BooleanField, ExpressionWrapper, Q, Case
 from django.http import Http404
+from django.utils.translation import gettext_lazy as _
 
 from PIL.JpegImagePlugin import JpegImageFile
-from PIL import ExifTags
+from PIL import ExifTags, Image, UnidentifiedImageError
+
+from photos.models import Photo
+
+logger = logging.getLogger(__name__)
 
 
 def photo_determine_rotation(pil_image):
```

Comment:
When reading the exif docs, the image could also be flipped. Those would be the even cases, I don't know if we want to support this tho.

Reply:
I think that this was an explicit choice, but since I'm not changing the functionality of the code here I do not think it is really relevant.

And when will we have flipped photos? Probably never.